games-emulation/retroarch/retroarch-1.7.1.recipe

```diff
@@ -0,0 +1,60 @@
+SUMMARY="The reference frontend of libretro"
+DESCRIPTION="RetroArch is a frontend for emulators, game engines and media players. \
+It enables you to run classic games on a wide range of computers and consoles \
+through its slick graphical interface. It has advanced features like shaders, \
+netplay, rewinding, next-frame response times, and more!"
+HOMEPAGE="https://libretro.com"
+COPYRIGHT="2010-2018 The RetroArch Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/libretro/RetroArch/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="e9189c7da6b2ca0142528eca14f26104ee322f052f8609203d90b60938d8d3f3"
+SOURCE_DIR="RetroArch-$portVersion"
+PATCHES="retroarch-$portVersion.patchset"
+
+ARCHITECTURES="x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	retroarch$secondaryArchSuffix = $portVersion
+	app:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch_cg2glsl$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	ffmpeg$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:which
+	cmd:git
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	"
+GLOBAL_WRITABLE_FILES="
+	$relativeSettingsDir/retroarch.cfg keep-old
+"
+
+BUILD()
+{
+	./fetch-submodules.sh
+	./configure --prefix=$installDestDir --global-config-dir=$settingsDir --bindir=$binDir --datarootdir=$dataRootDir --docdir=$docDir --mandir=$manDir --with-assets_dir=$dataDir
```

Comment:
```diff
+PATCH()
+{
+	libretroDir="`finddir B_SYSTEM_DIRECTORY`/$relativeLibDir/libretro"
+	retroarchDataDir="`finddir B_SYSTEM_DATA_DIRECTORY`/retroarch"
+	sed -i \
+		-e "s|^# \(libretro_directory =\)|\1 \"$libretroDir/\"|;" \
+		-e "s|^# \(libretro_info_path =\)|\1 \"$libretroDir/\"|;" \
+		-e "s|^# \(content_directory =\)|\1 \"$retroarchDataDir/\"|;" \
+		retroarch.cfg
+}
+
 BUILD()
 {
-	./fetch-submodules.sh
-	./configure --prefix=$installDestDir --global-config-dir=$settingsDir --bindir=$binDir --datarootdir=$dataRootDir --docdir=$docDir --mandir=$manDir --with-assets_dir=$dataDir
+	runConfigure --omit-dirs "binDir libExecDir infoDir" ./configure \
+		--bindir="$appsDir" --global-config-dir="$settingsDir" \
+		--with-assets_dir=$dataDir
 	make $jobArgs
 }
```
Regarding the `fetch-submodules.sh` script, it cannot be called by haikuporter because all the downloads need to be done by haikuporter itself.
This is going to be a bit complex, but we'll need to create as many `SOURCE_URI_2`, `SOURCE_URI_3`, etc... and their `CHECKSUM_SHA256_2`, `CHECKSUM_SHA256_3`, etc.. as there are submodules to download. Then we will have to use `$sourceDir2`, `$sourceDir3`, etc... to either create symlinks or copy these directories to the build directory.
(I'll guide you with this.)

Reply:
Sure, I'll start looking into this and will come back to you if I can't figure it out :D

Reply:
Also to keep it simple we could also just package these separately.

Reply:
No problem:)
BTW, in case you have a look at [haikuports/net-im/telegram-desktop](https://github.com/haikuports/haikuports/tree/master/net-im/telegram-desktop) please use `srcGitRev_2`, `srcGitRev_3`, etc... instead of `COMMIT`, which we try to avoid because we have far more recipes with `srcGitRev` than with `COMMIT`.